org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/Member.java

```diff
@@ -394,14 +395,31 @@ public boolean isBinary() {
 }
 protected boolean isMainMethod(IMethod method) throws JavaModelException {
 	if ("main".equals(method.getElementName()) && Signature.SIG_VOID.equals(method.getReturnType())) { //$NON-NLS-1$
-		int flags= method.getFlags();
+		int flags = method.getFlags();
+		ICompilationUnit icu = method.getCompilationUnit();
+		if (icu != null) {
+			Map<String, String> options = icu.getOptions(true);
+			if (JavaCore.ENABLED.equals(options.get(JavaCore.COMPILER_PB_ENABLE_PREVIEW_FEATURES))
+					&& JavaCore.VERSION_21.equals(options.get(JavaCore.COMPILER_CODEGEN_TARGET_PLATFORM))) {
+				if (!Flags.isPrivate(flags)) {
+					String[] paramTypes = method.getParameterTypes();
+					if (paramTypes.length == 1) {
+						String typeSignature = Signature.toString(paramTypes[0]);
```

Comment:
Could we extract this duplicate statements to a seriate method which checks weather the parameter is a String array ?